services/player-service/application/src/test/java/com/kjeldsen/player/application/usecases/GenerateSingleDeclineTrainingUseCaseTest.java

```diff
@@ -0,0 +1,88 @@
+package com.kjeldsen.player.application.usecases;
+
+import com.kjeldsen.events.domain.EventId;
+import com.kjeldsen.player.domain.Player;
+import com.kjeldsen.player.domain.repositories.PlayerReadRepository;
+import com.kjeldsen.player.domain.repositories.PlayerTrainingDeclineEventWriteRepository;
+import com.kjeldsen.player.domain.repositories.PlayerWriteRepository;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.time.Instant;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+
+public class GenerateSingleDeclineTrainingUseCaseTest {
+    private static final Integer DECLINE_SPEED = 50;
+    private static final Integer CURRENT_DAY = 4;
+    private static final Player.PlayerId PLAYER_ID = mock(Player.PlayerId.class);
+    private static final EventId EVENT_ID = EventId.from("event-id");
+    private static final Instant NOW = Instant.now();
+    final private PlayerWriteRepository mockedPlayerWriteRepository = Mockito.mock(PlayerWriteRepository.class);
+    final private PlayerReadRepository mockedPlayerReadRepository = Mockito.mock(PlayerReadRepository.class);
+    final private PlayerTrainingDeclineEventWriteRepository mockedPlayerTrainingDeclineEventWriteRepository = Mockito.mock(PlayerTrainingDeclineEventWriteRepository.class);
+    private final GenerateSingleDeclineTrainingUseCase generateSingleDeclineTrainingUseCaseMock =
+        new GenerateSingleDeclineTrainingUseCase(mockedPlayerTrainingDeclineEventWriteRepository,
+            mockedPlayerReadRepository, mockedPlayerWriteRepository);
+
+    @Test
+    @DisplayName("Generate inside GenerateSingleDeclineTrainingUseCase should throw an exception when player does not exist")
+    public void generate_Should_Throw_Exception_When_Player_Does_Not_Exist() {
+        // Arrange
+        when(mockedPlayerReadRepository.findOneById(null).isEmpty());
+
+        // Act & Assert
+        assertThrows(RuntimeException.class, () -> generateSingleDeclineTrainingUseCaseMock.generate(PLAYER_ID, CURRENT_DAY, DECLINE_SPEED));
+    }
+
+//    @Test
```

Comment:
pendiente de acabar?

Reply:
Pues si te digo la verdad parece ser que sí. Este es un test complicado que en su día empecé a hacer con Miguel (si mal no recuerdo) y no tengo mucha idea de por qué falla ni de como acabarlo...

También he puesto aquí (en el test correspondiente) el mensaje concreto para checkearlo ("Player not found.") que no me lo has puesto como comentario (imagino se te habrá pasado) pero que al igual que en los otros casos, es necesario.